### PR TITLE
[Cogs Face] Update Face SDK with latest auto-rest Swagger and update unit tests

### PR DESCRIPTION
**Pull request by lebronJ**
- Regenerate Face SDK with latest spec Azure/azure-rest-api-specs#2305
- Update unit tests (regenerate unit-test session records)
- Increment package version to `1.0.2`
- Divide `ComputerVision` and `Face` aligning with other Vision SDKs.
  
This has been internally reviewed (lebronJ/azure-sdk-for-net#1) inside Face SDK team.

### CODE REVIEW

File: src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Microsoft.Azure.CognitiveServices.Vision.csproj [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <PackageId>Microsoft.Azure.CognitiveServices.Vision</PackageId>
     <Description>This client library provides access to the Microsoft Cognitive Services Vision APIs.</Description>
-    <VersionPrefix>1.0.1-preview</VersionPrefix>
+    <VersionPrefix>1.0.2-preview</VersionPrefix>
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Microsoft.Azure.CognitiveServices.Vision.csproj]: @lebronJ please model your csproj according to the below project.
https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Microsoft.Azure.CognitiveServices.Search.CustomSearch.csproj
you are missing imports at the bottom of the project file and also you don't need few project properties that you have added.

File: src/SDKs/CognitiveServices/dataPlane/Vision/Vision.Tests/Face/BaseTests.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    public abstract class BaseTests
+    {
+        public static bool IsTestTenant = false;
+        private static readonly string FaceSubscriptionKey;
+        
+        static BaseTests()
+        {
+            // Retrieve the configuration information.
+            FaceSubscriptionKey = "";
+            Environment.SetEnvironmentVariable("AZURE_TEST_MODE", "Playback");
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/dataPlane/Vision/Vision.Tests/Face/BaseTests.cs]: @lebronJ you do not set this in test case, rather you set this in your connection string.
you use test framework to record tests
https://github.com/Azure/azure-powershell/blob/preview/documentation/Using-Azure-TestFramework.md#manually-set-environment-variables

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Please sync with David li in case you need more information about recording tests. If not let me know if you have any questions. I am tagging David on this PR

**Comment by shahabhijeet** [comment]
@DavidLiCIG can you review the test part of this PR

**Review by DavidLiCIG — APPROVED** [review]

File: src/SDKs/CognitiveServices/dataPlane/Vision/Vision.Tests/Face/BaseTests.cs (hunk 2 above)
Comment (lebronJ) [reply src/SDKs/CognitiveServices/dataPlane/Vision/Vision.Tests/Face/BaseTests.cs]: Removed this unnecessary setting
`Environment.SetEnvironmentVariable("AZURE_TEST_MODE", "Playback");`

File: src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Microsoft.Azure.CognitiveServices.Vision.csproj (hunk 1 above)
Comment (lebronJ) [reply src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Microsoft.Azure.CognitiveServices.Vision.csproj]: Refined this csproj following the styles of above `BingCustomSearch`

**Comment by lebronJ** [comment]
@shahabhijeet Updated following your comments.
If this is fine to merge, could you please help share how to release this updated Nuget package? Thanks.

**Comment by DavidLiCIG** [comment]
This package should be released as an update to Microsoft.Azure.CognitiveServices.Vision package which is already on version 1.0.1.preview.

File: src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Microsoft.Azure.CognitiveServices.Vision.csproj [hunk 3]
```diff
@@ -3,20 +3,15 @@
...
 
   <PropertyGroup>
     <TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
   </PropertyGroup>
 
-  <ItemGroup>
-    <Folder Include="Generated\Face\Models\" />
-    <Folder Include="Properties\" />
-  </ItemGroup>
+  <Import Condition=" Exists('$([MSBuild]::GetPathOfFileAbove(AzSdk.RP.props))') " Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.RP.props'))" />
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Microsoft.Azure.CognitiveServices.Vision.csproj]: @lebronJ add the following comment above/below line 2 and line 16

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Microsoft.Azure.CognitiveServices.Vision.csproj (hunk 3 above)
Comment (lebronJ) [reply src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Microsoft.Azure.CognitiveServices.Vision.csproj]: Done

**Comment by lebronJ** [comment]
@shahabhijeet Updated following your comments. Any updates? Thanks.

**Comment by shahabhijeet** [comment]
Have you used generate.cmd to generate your SDK? Using generate.cmd is essential to create the meta data file (it's a .txt file that has information about from what this SDK was generated)
I see your missing that file.

**Comment by lebronJ** [comment]
@shahabhijeet Updated by regenerating SDK with `generate.cmd`. Thanks for your comments and sorry for not noticing this before.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Please add your vision sdk under this directory
SDKs\CognitiveServices\dataPlane\Vision\VisionSdk\VisionSDKs\CognitiveServices\dataPlane\Vision\VisionSdk\Vision.Tests

This helps in identifying tests for your sdk.
This aligns with all the other SDKs that are checked in, e.g. ContentModerator

**Comment by lebronJ** [comment]
@shahabhijeet I have updated this PR by separating `Computer Vision` / `Face` SDKs and aligning them with `ContentModerator` and other Cogs SDKs using the same folder structure.
SDKs are regenerated by `generate.cmd` after adjusting the `readme.md` in Azure/azure-rest-api-specs#2305.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Please remove CognitiveServices/dataPlane/Vision/Vision/Generated/Face
I believe those files were left when you moved the files.

At this point you are better off create a new PR, this PR is already riddled with lot of commits.
It's better you create a new PR with all these changes with minimum commits

**Comment by lebronJ** [comment]
Thanks @shahabhijeet , I removed the unused Vision SDK folder as you commented. 
Considering creating a new PR with minimal commits, I think it's not straightforward to separate this PR, because this PR only does two **interdependent** things:
1. Regenerate Face & CV SDK with latest spec in Azure/azure-rest-api-specs#2305, and update unit tests based on the SDK changes.
2. Separate Face & CV SDKs aligning with other Vision SDKs.

So I rebased this PR to latest commit and force pushed it as one single commit.

**Review by shahabhijeet — APPROVED** [review]

**Comment by lebronJ** [comment]
Hi @shahabhijeet , could you please help share how to release this updated Nuget package? Thanks.
(Previously https://www.nuget.org/packages/Microsoft.Azure.CognitiveServices.Vision/)

One thing to note: 
As this PR totally separated `Computer Vision` and `Face` into two independent SDKs, I think we should release **TWO** Nuget packages `Microsoft.Azure.CognitiveServices.Vision.ComputerVision` and `Microsoft.Azure.CognitiveServices.Vision.Face`.

**Comment by DavidLiCIG** [comment]
Hi @lebronJ you can use this job to publish the new Nuget packages.

https://azuresdkci.westus2.cloudapp.azure.com/view/NetCore-SDK/job/NetCore-SDK-Publish/

**Comment by lebronJ** [comment]
Thanks @DavidLiCIG , published.

**Pull request merged**